lib/cf_app_discovery/client.rb

```diff
@@ -11,21 +11,42 @@ def apps
       paginator = Paginator.new do |next_url|
         get(next_url || "/v2/apps")
       end
-      paginator.flat_map do |page|
+
+      resources = paginator.flat_map do |page|
         page.fetch(:resources)
       end
+
+      resources.each do |resource|
+        resource[:route] = get_first_route(resource)
+      end
+      resources
     end
 
     def app(app_guid)
-      get("/v2/apps/#{app_guid}")
+      resource = get("/v2/apps/#{app_guid}")
+      resource[:route] = get_first_route(resource)
```

Comment:
this pattern (get a resource, then get_first_route on it) seems to be duplicated above. could we extract something common?

Reply:
sure, will refactor